core/src/main/java/com/netflix/iceberg/util/BinPacking.java

```diff
@@ -33,20 +33,22 @@
   public static class ListPacker<T> {
     private final long targetWeight;
     private final int lookback;
+    private final boolean expensiveTaskFirst;
 
-    public ListPacker(long targetWeight, int lookback) {
+    public ListPacker(long targetWeight, int lookback, boolean expensiveTaskFirst) {
       this.targetWeight = targetWeight;
       this.lookback = lookback;
+      this.expensiveTaskFirst = expensiveTaskFirst;
```

Comment:
I agree with this being a boolean option on the BaseTableScan but would it be more elegant to use a Comparator here instead of boolean so that ListPacker can be used for other possible sorting criteria?

Reply:
@prodeezy I like your suggestion. But I think it should be kept outside of this PR because it might involve changes on how we insert bins. Do you think if that's acceptable or do you want me to fold it into this PR?

Reply:
I see your point. Not entirely necessary for now methinks since it's the only sorting criteria right now.